src/rust/aggregator/src/pipelines/rolling_volume.rs

```diff
@@ -0,0 +1,67 @@
+use chrono::{DateTime, Duration, Utc};
+use sqlx::{PgConnection, PgPool};
+
+use aggregator::{util::*, Pipeline, PipelineAggregationResult};
+
+pub const TIMEOUT: std::time::Duration = std::time::Duration::from_secs(1);
+
+pub struct RollingVolume {
+    pool: PgPool,
+    last_indexed_timestamp: Option<DateTime<Utc>>,
+}
+
+impl RollingVolume {
+    pub fn new(pool: PgPool) -> Self {
+        Self {
+            pool,
+            last_indexed_timestamp: None,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Pipeline for RollingVolume {
+    fn model_name(&self) -> String {
+        String::from("RollingVolume")
+    }
+
+    fn ready(&self) -> bool {
+        self.last_indexed_timestamp.is_none()
+            || self.last_indexed_timestamp.unwrap() + Duration::from_std(TIMEOUT).unwrap()
+                < Utc::now()
+    }
+
+    async fn process_and_save_historical_data(&mut self) -> PipelineAggregationResult {
+        self.process_and_save_internal().await
+    }
+
+    fn poll_interval(&self) -> Option<std::time::Duration> {
+        Some(TIMEOUT)
+    }
+
+    async fn process_and_save_internal(&mut self) -> PipelineAggregationResult {
+        let mut transaction = create_repeatable_read_transaction(&self.pool).await?;
+        sqlx::query_file!(
+            "sqlx_queries/rolling_volume/insert_daily_rolling_volume.sql",
```

Comment:
Can we replace this pipeline with a view function that just gets the values from 24 hr candlesticks?

Reply:
We would lose a lot of precision.

The current code calculates the 24h volume for each minute. So that means we know the volume that was traded between 00h00 to 00h00 the next day, 00h01 to 00h01 the next day, et cetera.

Using candlesticks would only give us the volume between 00h00 to 00h00 the next day.

Having it minute to minute is useful to create graphs like this:

![image](https://github.com/econia-labs/econia/assets/67070716/f4b3ce66-5aee-4913-b63a-72e2c027d700)

rather than like this:

![image](https://github.com/econia-labs/econia/assets/67070716/9308d717-c418-428b-b522-25d99ff11f0a)